src/components/BaseCommentForm.js

```diff
@@ -45,273 +51,332 @@ Leaflet.Marker.prototype.options.icon = new Leaflet.Icon({
-      images = pluginComment.image ? [pluginComment.image] : images;
-      geojson = pluginComment.geojson || geojson;
-      pinned = pluginComment.pinned || null;
-      mapCommentText = pluginComment.mapCommentText || mapCommentText;
-    } else if (pluginData && typeof pluginData !== "string") {
+      setImages = pluginComment.image ? [pluginComment.image] : setImages;
+      setGeojson = pluginComment.geojson || setGeojson;
+      setPinned = pluginComment.pinned || null;
+      setMapCommentText = pluginComment.mapCommentText || setMapCommentText;
+    } else if (pluginData && typeof pluginData !== 'string') {
       // this is for old-fashioned plugins with only data
       pluginData = JSON.stringify(pluginData);
     }
 
     // validate form errors here before posting the comment
-    const userAnsweredAllQuestions = loggedIn && hasUserAnsweredAllQuestions(user, section);
+    const userHasAnsweredAllQuestions =
+      loggedIn && hasUserAnsweredAllQuestions(user, section);
+
     const errors = checkFormErrors(
-      this.state.imageTooBig, commentText, section, answers, isReply, userAnsweredAllQuestions
+      imageTooBig,
+      setCommentText,
+      section,
+      answers,
+      isReply,
+      userHasAnsweredAllQuestions
     );
+
     if (errors.length > 0) {
-      this.handleErrorStates(errors);
+      handleErrorStates(errors);
       return;
     }
 
     // make sure empty comments are not added when not intended
     if (isEmptyCommentAllowed(section, hasAnyAnswers(answers))) {
-      if (!commentText.trim()) {
-        commentText = config.emptyCommentString;
+      if (!setCommentText.trim()) {
+        setCommentText = config.emptyCommentString;
       }
     }
 
-    this.props.onPostComment(
-      commentText,
-      nickname,
+    onPostComment(
+      setCommentText,
+      setNickname,
       pluginData,
-      geojson,
+      setGeojson,
       label,
-      images,
-      pinned,
-      mapCommentText,
+      setImages,
+      setPinned,
+      setMapCommentText
     );
-    this.setState({
+
+    setFormData({
+      ...formData,
       collapsed: false,
-      commentText: "",
-      nickname: this.props.defaultNickname || '',
+      commentText: '',
+      nickname: defaultNickname || '',
       imageTooBig: false,
       images: [],
       pinned: false,
       showAlert: true,
       hideName: false,
       geojson: {},
-      mapCommentText: "",
+      mapCommentText: '',
       commentRequiredError: false,
       commentOrAnswerRequiredError: false,
     });
-  }
+  };
 
-  handleErrorStates(errors) {
-    this.setState({
-      commentRequiredError: errors.includes('commentRequiredError'),
-      commentOrAnswerRequiredError: errors.includes('commentOrAnswerRequiredError'),
-      imageTooBig: errors.includes('imageTooBig')
+  const isImageTooBig = (images) => {
+    let imageTooBig = false;
+    Array.from(images).forEach((image) => {
+      if (image.size > 1000000) {
```

Comment:
Optional opportunity to refactor this to be an environment variable from config.js

Reply:
https://github.com/City-of-Helsinki/kerrokantasi-ui/pull/958/commits/fe0e02c46761f286c0a3c2455f5dc45ba13c279a

Reply:
There's also https://github.com/City-of-Helsinki/kerrokantasi-ui/blob/master/src/config.js but upon further inspection I have no idea if and how it is populated